app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -402,7 +404,27 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
             // experiment on features close to startup.
             // But we need viaduct (the RustHttp client) to be ready before we do.
             components.analytics.experiments.initialize()
+
+            setupMessaging()
+        }
+    }
+
+    private fun setupMessaging() {
+        if (!FeatureFlags.messagingFeature) {
```

Comment:
@jhugman is it OK to remove the `Settings.isExperimentationEnabled` check?

Reply:
I see you're already checking `isExperimentationEnabled` in the `HomeFragment`, in which case I don't know what the desired behaviour is (if the user flips the studies switch, should they start seeing messages immediately, or on the next startup?).

This is a bit strange, but I added back in.

Reply:
It's a bit confusing. To make sure  users are in a consistent state when they turn on/off experiments, we let them know if they proceed the app will restart. 

<img width="458" alt="image" src="https://user-images.githubusercontent.com/773158/169049972-5dd81b36-7387-4430-99ba-20d57f536203.png">